whitesource/component.py

```diff
@@ -1,77 +1,84 @@
-import tempfile
+import tarfile
+import typing
 
-import ccc.github
-import whitesource.client
+import github3.github
+import github3.repos
+import dacite
+
+import sdo.labels
+import sdo.model
 
 import gci.componentmodel as cm
 
 
-def get_post_project_object(
-    whitesource_client: whitesource.client.WhitesourceClient,
-    component: cm.Component,
-    product_token: str,
-):
-    github_api = ccc.github.github_api_from_component(component=component)
-    github_repo = ccc.github.GithubRepo.from_component(component)
-
-    return PostProjectObject(
-        whitesource_client=whitesource_client,
-        github_api=github_api,
-        product_token=product_token,
-        component=component,
-        github_repo=github_repo,
-        component_version=component.version,
-    )
+def _get_ws_label_from_source(labels: typing.List[cm.Label]) -> sdo.labels.SourceIdHint:
+    for label in labels:
```

Comment:
consider https://github.com/gardener/component-spec/blob/master/bindings-python/gci/componentmodel.py#L105 :-)

Reply:
maybe we could add a `type`-attr to said find-function that one can pass a conversion-callable to also handle the deserialisation, wdyt?

Reply:
the function now uses the find_label method :)

Reply:
To handle the deserialisation inside the find_label function would be good